src/datastation/dv_dataverse_role_assignment.py

```diff
@@ -0,0 +1,127 @@
+    assignee = role_assignment.split('=')[0]
+    role = role_assignment.split('=')[1]
+    action = "None"
+    if in_current_assignments(assignee, role, dataverse_api):
+        print("{} is already {} for dataset {}".format(assignee, role, dataverse_api.get_alias()))
+    else:
+        print(
+            "Adding {} as {} for dataset {}".format(assignee, role, dataverse_api.get_alias()))
+        dataverse_api.add_role_assignment(assignee, role, dry_run=dry_run)
+        action = "Added"
+    csv_report.write(
+        {'DOI': dataverse_api.get_alias(), 'Modified': datetime.now(), 'Assignee': assignee, 'Role': role,
+         'Change': action})
+
+
+def in_current_assignments(assignee, role, dataverse_api: DataverseApi):
+    current_assignments = dataverse_api.get_role_assignments()
+    found = False
+    for current_assignment in current_assignments:
+        if current_assignment.get('assignee') == assignee and current_assignment.get(
+                '_roleAlias') == role:
+            found = True
+            break
+    return found
+
+
+def list_role_assignments(args, dataverse_client):
+    r = dataverse_client.dataverse(args.pid).get_role_assignments()
+    if r is not None:
+        rich.print_json(data=r)
+
+
+def remove_role_assignments(args, dataverse_client: DataverseClient):
+    pids = get_pids(args.pid_or_pid_file)
+    batch_processor = BatchProcessorWithReport(wait=args.wait, report_file=args.report_file,
+                                               headers=['alias', 'Modified', 'Assignee', 'Role', 'Change'])
+    batch_processor.process_pids(pids,
+                                 lambda pid,
+                                 csv_report: remove_role_assignment(args.role_assignment,
+                                                                    dataverse_api=dataverse_client.dataverse(pid),
+                                                                    csv_report=csv_report))
+
+def remove_role_assignment(role_assignment, dataverse_api: DataverseApi, csv_report, dry_run: bool = False):
+    assignee = role_assignment.split('=')[0]
+    role = role_assignment.split('=')[1]
+    action = "None"
+    if in_current_assignments(assignee, role, dataverse_api):
+        print("Removing {} as {} for dataverse {}".format(assignee, role, dataverse_api.get_alias()))
+        all_assignments = dataverse_api.get_role_assignments()
+        for assignment in all_assignments:
+            if assignment.get('assignee') == assignee and assignment.get('_roleAlias') == role:
+                dataverse_api.remove_role_assignment(assignment.get('id'), dry_run=dry_run)
+                action = "Removed"
+                break
+    else:
+        print("{} is not {} for dataset {}".format(assignee, role, dataverse_api.get_alias()))
+    csv_report.write(
+        {'alias': dataverse_api.get_alias(), 'Modified': datetime.now(), 'Assignee': assignee, 'Role': role,
+         'Change': action})
```

Comment:
Good point, I actually copied the code from `dv_dataset_role_assignment.py`, so the code is very similar and your suggestion would be a refactoring to prevent code duplication.

Reply:
It's a bit more complicated than I thought, the dataset_role_assigmenet uses the dataset_api and the datvaesre_role_assigment the dataverse_api...

Reply:
The code you mention has been moved to a DataverseRole class, but more refactoring would probably require to generalize the dataset and dataverse api, classes. Probably better to do this in a separate refactoring action